### PR TITLE
Update histogram plugin documentation

### PR DESCRIPTION
**Pull request by wchargin**
wchargin-branch: histogram-docs

### CODE REVIEW

**Comment by wchargin** [comment]
(Test failures are unrelated and appear to have been introduced by 2017-07-28 TensorFlow nightly; @chihuahua is looking into it.)

File: tensorboard/plugins/histogram/metadata.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 
 def create_summary_metadata(display_name, description):
+  """Create a `tf.SummaryMetadata` proto for histogram plugin data."""
```
Comment (chihuahua) [inline tensorboard/plugins/histogram/metadata.py]: Use a `Return:` block? A bit redundant, but maybe consistent and the return value shows up in HTML-based docs (if we ever generate them).

File: tensorboard/plugins/histogram/metadata.py [hunk 2]
```diff
@@ -40,4 +41,13 @@ def create_summary_metadata(display_name, description):
 
 
 def parse_summary_metadata(content):
+  """Parse summary metadata to a Python object.
+
+  Arguments:
+    content: the `content` field of a `SummaryMetadata` proto
```
Comment (chihuahua) [inline tensorboard/plugins/histogram/metadata.py]: pedantic nit: In other parts of this file, the description of each argument is capitalized and ends in a period.

Comment (wchargin) [inline tensorboard/plugins/histogram/metadata.py]: Thank you! I try to follow this convention but sometimes forget. I will fix this.

File: tensorboard/plugins/histogram/metadata.py (hunk 1 above)
Comment (wchargin) [inline tensorboard/plugins/histogram/metadata.py]: Sure; why not. :-)

**Comment by wchargin** [comment]
Thanks for the corrections; I agree with them. 

> shows up in HTML-based docs (if we ever generate them).

We certainly do want to hook ourselves into the tf docs generator. There's a way to do this: we need to instruct the doc generator to look at `tensorboard` as a top-level module in addition to `tf` and `tfdbg`.

**Pull request merged**